lib/ansible/constants.py

```diff
@@ -36,6 +40,26 @@ def _deprecated(msg, version):
         sys.stderr.write(' [DEPRECATED] %s, to be removed in %s\n' % (msg, version))
 
 
+def handle_config_noise(display=None):
+
+    if display is not None:
+        w = display.warning
```

Comment:
Since we're already accumulating config warnings and deprecations anyway, can we not just wait until `display` is functioning to show them instead of falling back to the pre-display emergency output? If we bomb out before display is up, config warnings and deprecation messages are probably not a primary concern anyway :laughing:.